packages/headless/src/features/facets/generic/facet-reducer-helpers.test.ts

```diff
@@ -42,37 +38,55 @@ describe('generic facet reducers', () => {
         const value = buildMockFacetValueRequest();
         const state = {[id]: buildMockFacetRequest({currentValues: [value]})};
 
-        handleFacetDeselectAll<FacetRequest>(state, id);
+        FacetReducers.handleFacetDeselectAll<FacetRequest>(state, id);
         expect(state[id].currentValues).toEqual([]);
       });
 
       it('sets #preventAutoSelect to true on the request', () => {
         const id = '1';
         const state = {[id]: buildMockFacetRequest({preventAutoSelect: false})};
 
-        handleFacetDeselectAll(state, id);
+        FacetReducers.handleFacetDeselectAll(state, id);
         expect(state[id].preventAutoSelect).toBe(true);
       });
     });
 
     it('when the passed id is not registered, it does not throw', () => {
-      expect(() => handleFacetDeselectAll({}, '1')).not.toThrow();
+      expect(() => FacetReducers.handleFacetDeselectAll({}, '1')).not.toThrow();
     });
   });
 
+  // it('#handleDeselectAllFacets calls #handleFacetDeselectAll for every facet', () => {
```

Comment:
In this test, I wanted to test that the `forEach` in `handleDeselectAllFacets` worked by spying on `handleFacetDeselectAll` which is call in each iteration. 

But this test fails because you can not mock `handleFacetDeselectAll`. I looked online for similar issues and I found a way to validate this test. In `facet-reducer-helpers.ts`, we would need to do a "self-import" (https://stackoverflow.com/a/55193363 under "Import the module into itself"). I tested that solution and it worked, but I don't know if this is a good idea. I wanted to get you input on this. 

The same issue applies for `range-facet-reducers.test.ts`

Reply:
Hmm, I see. I think creating a file for one function just for this purpose is not very nice. The self-import is more self-contained, and is less overhead if we need to repeat the pattern with other handler functions.

Reply:
So, should I do the self-import solution?

Reply:
Yep, I think this is the best approach,

Reply:
So Jenkins doesn't like that solution. ` Circular dependency`... Not sure what to do with this. 

As you said, creating a new file isn't very nice. I could change the test but I don't know to what. I could remove the test since it's only a loop and I tested manually with `console.log` in the function we are trying to spy on and it worked. So, I don't know if removing this test is ok. 🤔 

I don't know if it is possible to tell Jenkins to ignore the error for that file? 🤔

Reply:
What if we reverted to the original implementation that had some duplication of `Object.keys`?
If I had to pick between not having a test or having some duplication, I would personally choose the latter.

Reply:
Sounds good!